tests/unit/test-cmd/test.run.js

```diff
@@ -358,43 +226,31 @@ describe('run', () => {
       assert.equal(createdWatcher, watcher);
     });
 
-    it('reloads the extension', () => {
+    it('reloads the extension', async () => {
       const {config, createWatcher} = prepare();
       createWatcher();
 
       const callArgs = config.onSourceChange.firstCall.args[0];
       assert.typeOf(callArgs.onChange, 'function');
       // Simulate executing the handler when a source file changes.
-      return callArgs.onChange()
-        .then(() => {
-          assert.equal(config.addonReload.called, true);
-          const reloadArgs = config.addonReload.firstCall.args;
-          assert.ok(config.addonId);
-          assert.equal(reloadArgs[0].addonId, config.addonId);
-          assert.equal(reloadArgs[0].client, config.client);
-        });
+      await callArgs.onChange();
+      assert.equal(config.addonReload.called, true);
+      const reloadArgs = config.addonReload.firstCall.args;
+      assert.equal(reloadArgs[0], config.sourceDir);
     });
 
   });
 
   describe('defaultReloadStrategy', () => {
 
-    class StubChildProcess extends EventEmitter {
-      stderr = new EventEmitter();
-      stdout = new EventEmitter();
-      kill = sinon.spy(() => {});
-    }
-
-    function prepare() {
-      const client = new RemoteFirefox(fakeFirefoxClient());
+    function prepare({stubExtensionRunner} = {}) {
       const watcher = {
         close: sinon.spy(() => {}),
       };
+      const extensionRunner = createFakeExtensionRunner({}, {},
+                                                        stubExtensionRunner);
```

Comment:
I find code like this hard to read. I know it tends to be more verbose but I'd rather see named params passed like:

```js
createFakeExtensionRunner({
  params: {}, overiddenMethods: stubRunnerExtensionRunner,
});
```
In most cases here I don't think it would actually add much verbosity.